enocean/__init__.py

```diff
@@ -764,87 +764,60 @@ def send_switch_D2(self, id_offset=0, rx_id=0, pulsew=0, on=0):
             return None
-        else:
+        elif (device == 22):
+            # For Eltako FRGBW71L RGB dim devices (EEP 07-3F-7F)
+            payload = [0xFF, 0xF8, 0x0D, 0x87]
             self.logger.info('enocean: sending learn telegram for rgbw dim command with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
-            self._send_radio_packet(id_offset, rorg, payload)
-            return None
-
-
-    def send_learn_switch(self, id_offset=0):
-        #self.logger.debug('enocean: send_learn_switch()')
-        # Prepare Data
-        rorg = 0xA5
-        payload = [0xE0, 0x40, 0x0D, 0x80]
-
-        if (id_offset < 0) or (id_offset > 127):
-            self.logger.error('enocean: ID offset with value = {} out of range (0-127). Aborting.'.format(id_offset))
             return None
-        else:
-            self.logger.info('enocean: sending learn telegram for switch command with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
-            self._send_radio_packet(id_offset, rorg, payload)
-            return None
-
-
-    def send_learn_radiator_valve(self, id_offset=0):
-        #self.logger.debug('enocean: send_learn_radiator_valve()')
-        # Prepare Data
-        rorg = 0xA5
-        payload = [0x00, 0x00, 0x00, 0x00]
-
-        if (id_offset < 0) or (id_offset > 127):
-            self.logger.error('enocean: ID offset with value = {} out of range (0-127). Aborting.'.format(id_offset))
+        # device range 30 - 39 --> Learn protocol for radiator valves
+        elif (device == 30):
+            # Radiator Valve
+            payload = [0x00, 0x00, 0x00, 0x00]
+            self.logger.info('enocean: sending learn telegram for radiator valve with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
             return None
+        # device range 40 - 49 --> Learn protocol for other actuators
+        elif (device == 40):
+            # Eltako shutter actor FSB14, FSB61, FSB71
+             payload = [0xFF, 0xF8, 0x0D, 0x80]
+             self.logger.info('enocean: sending learn telegram for actuator with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
+             return None
         else:
-            self.logger.info('enocean: sending learn telegram for radiator valve with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
-            self._send_radio_packet(id_offset, rorg, payload)
+            self.logger.error('enocean: no such device defined for learning!')

```

Comment:
Maybe this error message could include the missing device so that users see this in the log and can report it somewhere.